### PR TITLE
sql: move copy_in_test to pkg/sql/copy to have all copy tests together

### PR DESCRIPTION
**Pull request by cucaroach**
Epic: CRDB-18892
Informs: #91831
Release note: None

### CODE REVIEW

File: pkg/sql/copy/copy_in_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-package sql_test
+package copy_test
```
Comment (otan) [inline pkg/sql/copy/copy_in_test.go:11]: this can just be `package copy` now right

File: pkg/sql/sqltestutils/sql_test_utils.go [hunk 2]
```diff
@@ -140,3 +142,13 @@ func ArrayStringToSlice(t *testing.T, array string, message ...string) []string
 	}
 	return strings.Split(array[1:length-1], ",")
 }
+
+// PgxConn is a helper to create a pgx.Conn from a url.
+func PgxConn(t *testing.T, connURL url.URL) (*pgx.Conn, error) {
```
Comment (otan) [inline pkg/sql/sqltestutils/sql_test_utils.go:147]: nit: `PGXConn` is the best practice form

**Review by otan — APPROVED** [review]

File: pkg/sql/copy/copy_in_test.go (hunk 1 above)
Comment (cucaroach) [reply pkg/sql/copy/copy_in_test.go:11]: Umm, IDK, what's the motivation? Is the idea to use _test only when its necessary to break a package dep cycle?    I thought _test was good hygiene for the only test/use the public bits perspective (which I don't really buy but not the hill I want to die on).   Curious your thoughts...

Comment (otan) [reply pkg/sql/copy/copy_in_test.go:11]: > _test only when its necessary to break a package dep cycle

that was my understanding. that and it has caused me headaches in the past when doing build work :P but not important here.

>  I thought _test was good hygiene for the only test/use the public bits perspective

eh, usually for tests you want to do unit tests with right

----

either way, not a hill i'm looking to die on either.

Comment (cucaroach) [reply pkg/sql/copy/copy_in_test.go:11]: I'm gonna go with if you remove _test and everything still works then it shouldn't be there as a good principle here.  Thanks!

**Comment by cucaroach** [comment]
TFTRs!
bors r+

**Pull request merged**